app/ts/store/balance.ts

```diff
@@ -0,0 +1,48 @@
+import { signal } from '@preact/signals'
+import { ethers } from 'ethers'
+import { assertsExternalProvider } from '../library/utilities.js'
+import { accountStore } from './account.js'
+
+type BalanceStore =
+	| {
+			state: 'inactive'
+			checkEthBalance: () => Promise<void>
+	  }
+	| {
+			state: 'pending'
+			reset: () => void
+	  }
+	| {
+			state: 'rejected'
+			error: Error
+			reset: () => void
+	  }
+	| {
+			state: 'resolved'
+			balance: string
+			lastChecked: Date
+			reset: () => void
+	  }
```

Comment:
This code *looks* a lot like `useAsyncState`.  I'm curious what the difference is, and why you chose to roll this instead of using `useAsyncState`?

Reply:
#54 will be working on this soon

Reply:
This will be applied on #70